packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -3067,6 +3067,36 @@ export class ContainerRuntime
 						error,
 					};
 				}
+
+				// If there are pending messages, the summary has more data than at the summaryRefSeqNum.
+				if (this.hasPendingMessages()) {
+					// If "SkipFailingIncorrectSummary" option is true, don't fail the summary in the last attempt.
+					// This is a fallback to make progress in documents where there are consistently pending ops in
+					// the summarizer.
+					if (
+						finalAttempt &&
+						this.mc.config.getBoolean("Fluid.Summarizer.SkipFailingIncorrectSummary")
```

Comment:
i think this should be on be default to start with

Reply:
i would also maybe log as a data processing error, as it can lead to corruption

Reply:
Yeah, this should be a data processing error.

Reply:
We explicitly chose to not make this the default because we want to keep the defaults clean and any bad patterns to be explictly opted in. Also, from past experience, its harder to enabled a feature flag that would opt out of this behavior.

Reply:
>> Having said that, the plan is to have both in place before any of this is enabled.

if you check this in as is today, it will be released tomorrow, and could be consumer by any partner, which there are many, i don't think from the plan laid out this change is ready for that.

the plan would be the flip the default later, not have partners set the config, breaking summaries is not a good option.

Reply:
This is part of a larger plan to add increased validation and retries to summarization. Both of them are behind feature flags which are not enabled by default. This logic for instance is under `this.validateSummaryBeforeUpload` check which is not enabled by default.
We will test it in stages, look at the data and then decide which parts to enable by default and which need more work before doing so.

We have got to start somewhere, otherwise we will never get out of this state where there are local changes in summarizer which keep corrupting documents, and we spent days trying to figure things out fix issues. We have been in this mode for the last couple of years.
I am not saying the answer is corrupting documents by failing summarization as that is definitely not the plan.

Reply:
ok. i think i may have misunderstood the set of flags. maybe we want to set them in stress, which should hit it, but then at least the code would rung

Reply:
Good point. The validate flag is already turned on as both on and off. Same for the flag that uses the new retry logic. I will add this one as well. That reminds me I should check the logs to see about the data for the other two.

Reply:
I will turn the feature flag for this one on when I add tests for it. I was planning to add tests to this PR itself but it requires a bunch of other changes. So, maybe I will wait to merge this while I work on the other PR to set up for tests.

Reply:
AB#5364